src/makeEslinter.ts

```diff
@@ -0,0 +1,50 @@
+import * as eslinttypes from 'eslint'; // import for types alone
+import { NormalizedMessage } from './NormalizedMessage';
+import { FsHelper } from './FsHelper';
+import { makeCreateNormalizedMessageFromEsLintFailure } from './NormalizedMessageFactories';
+
+export function makeEslinter(eslintOptions: object) {
+  // See https://eslint.org/docs/1.0.0/developer-guide/nodejs-api#cliengine
+  const eslint: typeof eslinttypes = require('eslint');
+  const eslinter = new eslint.CLIEngine(eslintOptions);
+  const createNormalizedMessageFromEsLintFailure = makeCreateNormalizedMessageFromEsLintFailure();
+
+  function getLintsForFile(filepath: string) {
+    try {
+      const lints = eslinter.executeOnFiles([filepath]);
+      return lints;
+    } catch (e) {
+      if (
+        FsHelper.existsSync(filepath) &&
+        // check the error type due to file system lag
+        !(e instanceof Error) &&
```

Comment:
I saw this code somewhere else - can we extract it to the function? I would call it `isInvalidSourceFileError` or similar

Reply:
yeah maybe